src/internal/kopia/wrapper.go

```diff
@@ -366,6 +366,36 @@ func (w Wrapper) RestoreMultipleItems(
 		cols = map[string]*kopiaDataCollection{}
 	)
 
+	// TODO(meain): We have to update the details that is getting
+	// stored to only contain the filenames, as of now it contains
+	// every file that gets saved. Once we do this, we will have to
+	// have the component augment the list of the files here while
+	// restoring to make sure we add back in all the data and metadata
+	// files that we need to form the collections.
+
+	// NB: We have to make sure to order the files that gets fetched
+	// by kopia to have it pull the metadata file for the folder
+	// first, and for the metadata files for the file to be pulled
+	// right after the file for it to be pulled.
+
+	// TODO(meain) If we end converting the reads to parallel reads,
+	// how would it affect files showing up at the other side or
+	// should be make kopia guarantee that it will send us the file in
+	// the order that we send the details list in.
+
+	// TODO(meain): details things can avoid storing info about
+	// .dirmeta and we can compute them from the list of files. The
+	// only issue is that we might not be able to restore empty folder
+	// as we will not be aware of them without some mention about
+	// them.
+
+	// This sort is done primarily to order .meta files after .data
+	// files
+	// TODO(meain): This code should be moved into the OneDrive component
+	sort.Slice(paths, func(i, j int) bool {
```

Comment:
One thing we still rely on is the fact that we get the stream items with the item before its metadata. It does not matter if we get all the items first, then all the the items, but we cannot handle if we get the metadata first before the actual item data. Hope this is good.

Reply:
yea, I was kind of thinking about this after our chat. Eventually I'd like to remove the ordering requirements from kopia wrapper as it constrains the possible implementations we can choose and how things will work when we may not know what files need selected beforehand (e.x. restoring an email with attachments)

For now I can't think of a different solution though :/ maybe others will have some thoughts